INSTALL.md

```diff
@@ -0,0 +1,236 @@
+
+      pip install .
+
+
+**You can also install recommended version of the dependencies by using
+constraint- ```python<PYTHON_MAJOR_MINOR_VERSION>.txt``` files as constraint file. This is needed in case
+you have problems with installing the current requirements from PyPI.
+There are different constraint files for different python versions. For example:**
+
+      pip install . \
+        --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-main/constraints-3.8.txt"
+
+
+By default `pip install` in Airflow 2.0 installs only the provider packages that are needed by the extras and
+install them as packages from PyPI rather than from local sources:
+
+      pip install .[google,amazon] \
+        --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-main/constraints-3.8.txt"
+
+
+You can upgrade just airflow, without paying attention to provider's dependencies by using 'constraints-no-providers'
+constraint files. This allows you to keep installed provider packages.
+
+      pip install . --upgrade \
+        --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-main/constraints-no-providers-3.8.txt"
+
+
+You can also install airflow in "editable mode" (with `-e`) flag and then provider packages are
+available directly from the sources (and the provider packages installed from `PyPI` are UNINSTALLED in
+order to avoid having providers in two places. And `provider.yaml` files are used to discover capabilities
+of the providers which are part of the airflow source code.
+
+You can read more about `provider.yaml` and community-managed providers in
+`https://airflow.apache.org/docs/apache-airflow-providers/index.html` for developing custom providers
+and in ``CONTRIBUTING.rst`` for developing community maintained providers.
+
+This is useful if you want to develop providers:
+
+      pip install -e . \
+        --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-main/constraints-3.8.txt"
+
+You can also skip installing provider packages from `PyPI` by setting `INSTALL_PROVIDERS_FROM_SOURCE` to "true".
+In this case Airflow will be installed in non-editable mode with all providers installed from the sources.
+Additionally `provider.yaml` files will also be copied to providers folders which will make the providers
+discoverable by Airflow even if they are not installed from packages in this case.
+
+      INSTALL_PROVIDERS_FROM_SOURCES="true" pip install . \
+        --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-main/constraints-3.8.txt"
+
+Airflow can be installed with extras to install some additional features (for example 'async' or 'doc' or
+to install automatically providers and all dependencies needed by that provider:
+
+      pip install .[async,google,amazon] \
+        --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-main/constraints-3.8.txt"
+
+The list of available extras:
```

Comment:
Are extras still recommended for deployment? IIRC they are mostly kept only for compatibility, and for an actual production installation it is recommended to install the actual providers you want instead, which would pull in the dependencies themselves.